lib/navigation/imu_displacement.cpp

```diff
@@ -0,0 +1,55 @@
+#include "imu_displacement.hpp"
+
+#include <cmath>
+
+#include <chrono>
+
+namespace hyped::navigation {
+
+ImuDisplacement::ImuDisplacement(const core::ITimeSource &time)
+    : time_(time),
+      imu_displacement_(0),
+      imu_velocity_(0)
+{
+}
+
+void ImuDisplacement::updateImuDisplacement(const core::Float imu_acceleration)
+{
+  const core::TimePoint time_now = time_.now();
+  core::Timer timer_(time_);
+  const core::Duration time_elapsed = timer_.measure_lapsed_time(previous_timestamp_);
+  const core::Float time_elapsed_seconds
+    = std::chrono::duration_cast<std::chrono::microseconds>(time_elapsed).count() / 1'000'000;
```

Comment:
i believe i added a helper for this in `time.hpp` too, possibly on an unmerged branch (see above)

if not, make one

we don't want this ugly chrono syntax anywhere outside `core`